frame/support/src/lib.rs

```diff
@@ -222,6 +222,37 @@ macro_rules! assert_ok {
 	}
 }
 
+/// Opens a new tracing span. A no-op in WASM.
+///
+/// The returned span can be later entered into using `tracing_span_enter`.
```

Comment:
```suggestion
/// The returned span can be later entered into using `tracing_span_enter`.
```
Note it's not actually returning the span, which is OK for now because we don't currently support recording fields after span creation. However perhaps it's possible to allow input to the macro to record fields; as in: https://tracing.rs/tracing/#recording-fields

Reply:
Yeah, sorry. The docs are left-overs from my previous approach, where I wanted to make it a bit more flexible to use.
However I got stuck somewhere between the `span` vs `guard` lifetime issues and macro magic with multiple `tt` types and ended up with a simplistic approach to just pass the `span` name and the code without any configuration options.

I think if needed it can be expanded later on.